pyproject.toml

```diff
@@ -55,6 +55,17 @@ cov = [
   "cov-report",
 ]
 
+[tool.hatch.envs.hatch-test]
+dependencies = [
```

Comment:
Parts of these sections are duplicated. Can we just have a single reference?

Reply:
Yes this a good point. We can avoid the duplicates by templating, e.g.,
```
[tool.hatch.envs.hatch-test]
dependencies = [
  "coverage[toml]>=6.5; python_version >= '3.8'",
  "pytest",
  "flask",
  "flask-jwt-extended",
]

[tool.hatch.envs.default]
template = "hatch-test"
```

But this is not supported by earlier hatch version 1.7.0, which is run under python 3.7. Or I lack the knowledge of doing this so please feel free to suggest.

This early hatch version is currently used by `docker-hatch`. If we decide that we don't need it anymore, we can refactor it here.

Reply:
`docker-hatch` is simply a way to run any hatch command inside a consistent docker image. 

No need for people to install python & hatch locally...

`docker-hatch` also allows you to quickly run a command for various python version. This is useful to help debug commands when you see them failing on ci/cd for a specific version of python (e.g. ./docker-hatch -v 3.9 run lint:all): https://github.com/Kaggle/kagglehub/blob/c0de979194b2e9a623644cf7021c47953ecc0ff6/docker-hatch#L25-L29

Reply:
@rosbo So I understand currently `docker-hatch` works as this:
1. We first install the python with specific `PYTHON_VERSION`.
2. We install hatch that works with the current python version.

The challenge of step 2 is: 
1. we will need to install different hatch versions based on whether py <= 3.7 or not: hatch==1.7.0 for py<=3.7 and latest hatch for py > 3.7.
2. When we install different hatch version, `hatch test` doesn't work out of box for hatch <= 1.10.0, so we will still need `script test`. 

It seems we have different options:
1. Removing docker-hatch`, and use the latest hatch version (e.g., 1.12.0). Testing and linting for different python versions are supported this way (see PR summary above).
2. Having conditions for hatch installation, and run different commands `hatch test` vs `hatch run test` for python 3.7 vs later, e.g.,

```
ARG PYTHON_VERSION

FROM python:${PYTHON_VERSION}

RUN if [ "$(printf '%s\n' "3.7" "${PYTHON_VERSION}" | sort -V | head -n 1)" != "${PYTHON_VERSION}" ]; then \
        pip install "hatch==1.7.0"; \
    else \
        pip install "hatch==1.12.0"; \
    fi
```

What do you think in this case?